### ISSUE TITLE
Failed to run CLI from MacOS app bundle

### ISSUE THREAD

**Issue by asl** [asl/BandageNG#99]
There seems to be more to this than just inherited bugs.
I can start both the "original" Bandage as well as BandageNG from the CLI, and I can also display the version numbers of both of them:
```
$ /Applications/Bandage.app/Contents/MacOS/Bandage --version 
Version: 0.9.0
$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --version
Version: 0.9.6-dev
```
But other options dont seem to work with BandageNG:

```
$ /Applications/Bandage.app/Contents/MacOS/Bandage --help

  ____                  _                  
 |  _ \                | |                 
 | |_) | __ _ _ __   __| | __ _  __ _  ___ 
 |  _ < / _` | '_ \ / _` |/ _` |/ _` |/ _ \
 | |_) | (_| | | | | (_| | (_| | (_| |  __/
 |____/ \__,_|_| |_|\__,_|\__,_|\__, |\___|
                                 __/ |     
                                |___/      
Version: 0.9.0

Usage:    Bandage <command> [options]
          
Commands: <blank>      Launch the Bandage GUI
          load         Launch the Bandage GUI and load a graph file
          info         Display information about a graph
          image        Generate an image file of a graph
          querypaths   Output graph paths for BLAST queries
          reduce       Save a subgraph of a larger graph
          
Options:  --help       View this help message
          --helpall    View all command line settings
          --version    View Bandage version number
          
Online Bandage help: https://github.com/rrwick/Bandage/wiki









          
$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --help
qt.qpa.plugin: Could not find the Qt platform plugin "minimal" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa.

zsh: abort      /Applications/BandageNG.app/Contents/MacOS/BandageNG --help
```

or as another example:

```
$ /Applications/Bandage.app/Contents/MacOS/Bandage image assembly.gfa test.png --colour uniform
--> succesfully creates the image


$ /Applications/BandageNG.app/Contents/MacOS/BandageNG image assembly.gfa test.png --colour uniform
qt.qpa.plugin: Could not find the Qt platform plugin "minimal" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa.

zsh: abort      /Applications/BandageNG.app/Contents/MacOS/BandageNG image   --colour uniform
```

EDIT:
This behaviour is also present in the new release
```
$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --version      
Version: 2022.6-dev

$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --help   
qt.qpa.plugin: Could not find the Qt platform plugin "minimal" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa.

zsh: abort      /Applications/BandageNG.app/Contents/MacOS/BandageNG --help
```

_Originally posted by @richardstoeckl in https://github.com/asl/BandageNG/issues/47#issuecomment-1190159823_

**Comment by asl** [comment]
@richardstoeckl current workaround is to add `-platform cocoa` to cmdline

**Comment by richardstoeckl** [comment]
> @richardstoeckl current workaround is to add `-platform cocoa` to cmdline

Sorry, but I dont quite understand this workaround:

Without changing anything, CLI options like `--help` dont seem to work with BandageNG:
```
$ /Applications/Bandage.app/Contents/MacOS/Bandage --help

  ____                  _                  
 |  _ \                | |                 
 | |_) | __ _ _ __   __| | __ _  __ _  ___ 
 |  _ < / _` | '_ \ / _` |/ _` |/ _` |/ _ \
 | |_) | (_| | | | | (_| | (_| | (_| |  __/
 |____/ \__,_|_| |_|\__,_|\__,_|\__, |\___|
                                 __/ |     
                                |___/      
Version: 0.9.0

Usage:    Bandage <command> [options]
          
Commands: <blank>      Launch the Bandage GUI
          load         Launch the Bandage GUI and load a graph file
          info         Display information about a graph
          image        Generate an image file of a graph
          querypaths   Output graph paths for BLAST queries
          reduce       Save a subgraph of a larger graph
          
Options:  --help       View this help message
          --helpall    View all command line settings
          --version    View Bandage version number
          
Online Bandage help: https://github.com/rrwick/Bandage/wiki

--------------------------------------------------------------------------------------------------------
          
$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --help
qt.qpa.plugin: Could not find the Qt platform plugin "minimal" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa.

zsh: abort      /Applications/BandageNG.app/Contents/MacOS/BandageNG --help
```

When I add `-platform cocoa`, I can get the help message to show:
```
$ /Applications/BandageNG.app/Contents/MacOS/BandageNG --help -platform cocoa

  ____                  _                  
 |  _ \                | |                 
 | |_) | __ _ _ __   __| | __ _  __ _  ___ 
 |  _ < / _` | '_ \ / _` |/ _` |/ _` |/ _ \
 | |_) | (_| | | | | (_| | (_| | (_| |  __/
 |____/ \__,_|_| |_|\__,_|\__,_|\__, |\___|
                                 __/ |     
                                |___/      
Version: 2022.6-dev

Usage:    BandageNG <command> [options]
          
Commands: <blank>      Launch the Bandage-NG GUI
          load         Launch the Bandage-NG GUI and load a graph file
          info         Display information about a graph
          image        Generate an image file of a graph
          querypaths   Output graph paths for BLAST queries
          reduce       Save a subgraph of a larger graph
          
Options:  --help       View this help message
          --helpall    View all command line settings
          --version    View Bandage version number
          
Online Bandage help: https://github.com/asl/BandageNG/wiki
```

but other commands such as `image` do not seem to work:
```
$ /Applications/Bandage.app/Contents/MacOS/Bandage image "assembly.gfa" "test.png"
--> successfully creates image

--------------------------------------------------------------------------------------------------------

$ /Applications/BandageNG.app/Contents/MacOS/BandageNG image "assembly.gfa" "test.png"
qt.qpa.plugin: Could not find the Qt platform plugin "minimal" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa.

zsh: abort      /Applications/BandageNG.app/Contents/MacOS/BandageNG image  "test.png"


--------------------------------------------------------------------------------------------------------

$ /Applications/BandageNG.app/Contents/MacOS/BandageNG image "assembly.gfa" "test.png" -platform cocoa

Bandage-NG error: Invalid options: -platform, cocoa
```

Am I doing something wrong?

**Comment by asl** [comment]
No, you did not. Sadly, CLI argument handling is a bit flawed and non-flexible (#50 is a task to fix this). The merged pull request (#100) fixed the underlying issue. So it will be available soon as a part of next release / snapshot. https://github.com/asl/BandageNG/issues/47#issuecomment-1191390759 contains an explanation of what is going on.

I will likely make a pre-release as soon as linux issue will be fixed upstream

**Comment by richardstoeckl** [comment]
Ah sorry, I somehow missed the line about my issue in the comment there. Thank you so much for your time and effort, and for your kind explanations!
